__tests__/integration/hathorwallet_facade.test.ts

```diff
@@ -34,11 +34,62 @@ import Network from '../../src/models/network';
 import { WalletType } from '../../src/types';
 import { parseScriptData } from '../../src/utils/scripts';
 import { MemoryStore, Storage } from '../../src/storage';
+import { TransactionTemplateBuilder } from '../../src/template/transaction';
 
 const fakeTokenUid = '008a19f84f2ae284f19bf3d03386c878ddd15b8b0b604a3a3539aa9d714686e1';
 const sampleNftData =
   'ipfs://bafybeiccfclkdtucu6y4yc5cpr6y3yuinr67svmii46v5cfcrkp47ihehy/albums/QXBvbGxvIDEwIE1hZ2F6aW5lIDI3L04=/21716695748_7390815218_o.jpg';
 
+describe('template methods', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should build transactions from the template transaction', async () => {
+    const hWallet = await generateWalletHelper();
+    const address = await hWallet.getAddressAtIndex(1);
+
+    await GenesisWalletHelper.injectFunds(hWallet, address, 10n);
+
+    const template = new TransactionTemplateBuilder()
+      .addConfigAction({ tokenName: 'Tmpl Token', tokenSymbol: 'TT' })
+      .addSetVarAction({ name: 'addr', call: { method: 'get_wallet_address' } })
+      .addUtxoSelect({ fill: 1 })
+      .addTokenOutput({ address: '{addr}', amount: 100, useCreatedToken: true })
+      .build();
+
+    const tx = await hWallet.buildTxTemplate(template, { signTx: true, pinCode: DEFAULT_PIN_CODE });
+    expect(tx.version).toEqual(2); // Create token transaction
+    expect(tx.inputs).toHaveLength(1);
+    expect(tx.inputs[0].data).not.toBeFalsy(); // Tx is signed
+    // Transaction is not mined yet
```

Comment:
suggestion(non-blocking): why don't you mine and send this now to complete the test?

Reply:
Added the expect, but the if statement is required to ensure typing consistency